app/routes/jobs.$id._index.tsx

```diff
@@ -40,11 +43,41 @@ export const action = async ({ params, request }: ActionFunctionArgs) => {
 
 export default function JobPage() {
   const { job } = useLoaderData<typeof loader>();
+  const { revalidate } = useRevalidator();
+
+  useEffect(() => {
+    // set interval to refresh job status
+    const id = setInterval(() => {
+      // reload only if user is on the page/tab
+      if (
+        document.visibilityState === "visible" &&
```

Comment:
Nice, did not know that property existed.  I was hoping it would also stop when another Window is covering it, but that did not stop it refreshing, but minimizing or not being active tab did.